show/bgp_frr_v4.py

```diff
@@ -24,43 +24,93 @@ def bgp():
 @bgp.command()
 @multi_asic_util.multi_asic_click_options
 def summary(namespace, display):
-    bgp_summary = bgp_util.get_bgp_summary_from_all_bgp_instances(constants.IPV4, namespace,display)
+    bgp_summary = bgp_util.get_bgp_summary_from_all_bgp_instances(
+        constants.IPV4, namespace, display)
     bgp_util.display_bgp_summary(bgp_summary=bgp_summary, af=constants.IPV4)
-    
+
 
 # 'neighbors' subcommand ("show ip bgp neighbors")
 @bgp.command()
 @click.argument('ipaddress', required=False)
-@click.argument('info_type', type=click.Choice(['routes', 'advertised-routes', 'received-routes']), required=False)
-def neighbors(ipaddress, info_type):
+@click.argument('info_type',
+                type=click.Choice(
+                    ['routes', 'advertised-routes', 'received-routes']),
+                required=False)
+@click.option('--namespace',
+                '-n',
+                'namespace',
+                default=None,
+                type=str,
+                show_default=True,
+                help='Namespace name or all',
+                callback=multi_asic_util.multi_asic_namespace_validation_callback)
+def neighbors(ipaddress, info_type, namespace):
     """Show IP (IPv4) BGP neighbors"""
 
-    command = 'sudo vtysh -c "show ip bgp neighbor'
-
+    command = 'show ip bgp neighbor'
     if ipaddress is not None:
-        command += ' {}'.format(ipaddress)
+        if not bgp_util.is_ipv4_address(ipaddress):
+            ctx = click.get_current_context()
+            ctx.fail("{} is not valid ipv4 address\n".format(ipaddress))
+        try:
+            actual_namespace = bgp_util.get_namespace_for_bgp_neighbor(
+                ipaddress)
+            if namespace is not None and namespace != actual_namespace:
+                click.echo(
+                    "[WARNING]: bgp neighbor {} is present in namespace {} not in {}"
+                    .format(ipaddress, actual_namespace, namespace))
 
-        # info_type is only valid if ipaddress is specified
-        if info_type is not None:
-            command += ' {}'.format(info_type)
+            # save the namespace in which the bgp neighbor is configured
+            namespace = actual_namespace
+
+            command += ' {}'.format(ipaddress)
 
-    command += '"'
+            # info_type is only valid if ipaddress is specified
+            if info_type is not None:
+                command += ' {}'.format(info_type)
+        except ValueError as err:
+            ctx = click.get_current_context()
+            ctx.fail("{}\n".format(err))
+
+    ns_list = multi_asic.get_namespace_list(namespace)
+    output = ""
+    for ns in ns_list:
```

Comment:
Question on the overall design to handle multi-ASIC.  By looking at how this is implemented it looks like it will cycle through all the name spaces and no attempt to merge/filter but will display whatever each namespace has to the output directly?

Reply:
@gechiang, Yes, this command we just display neighbor from all the namespaces. No filtering done for this commands